src/Compilers/Core/CodeAnalysisTest/CryptoBlobParserTests.cs

```diff
@@ -3,21 +3,93 @@
 using Roslyn.Test.Utilities;
 using System;
 using System.Collections.Immutable;
+using System.Linq;
 using System.Diagnostics;
 using System.Globalization;
 using Xunit;
+using System.Security.Cryptography;
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
     public class CryptoBlobParserTests : TestBase
     {
         [Fact]
+        public void GetPrivateKeyFromKeyPair()
+        {
+            var key = ImmutableArray.Create(TestResources.General.snKey);
+
+            RSAParameters? privKeyOpt;
+            Assert.True(CryptoBlobParser.TryParseKey(key, out _, out privKeyOpt));
+            Debug.Assert(privKeyOpt.HasValue);
+            var privKey = privKeyOpt.Value;
+
+            Assert.Equal(privKey.Exponent, new byte[] { 0x01, 0x00, 0x01, 0x00 });
+
+            var expectedModulus = key.Skip(20).Take(128).ToArray();
+            Assert.Equal(expectedModulus, privKey.Modulus);
+            
+            var expectedP = key.Skip(20 + 128).Take(64).ToArray();
+            Assert.Equal(expectedP, privKey.P);
+
+            var expectedQ = key.Skip(20 + 128 + 64).Take(64).ToArray();
+            Assert.Equal(expectedQ, privKey.Q);
+
+            var expectedDP = key.Skip(20 + 128 + 64 + 64).Take(64).ToArray();
+            Assert.Equal(expectedDP, privKey.DP);
+
+            var expectedDQ = key.Skip(20 + 128 + 64 + 64 + 64).Take(64).ToArray();
+            Assert.Equal(expectedDQ, privKey.DQ);
+
+            var expectedInverseQ = key.Skip(20 + 128 + 64 + 64 + 64 + 64).Take(64).ToArray();
+            Assert.Equal(expectedInverseQ, privKey.InverseQ);
+
+            var expectedD = key.Skip(20 + 128 + 64 + 64 + 64 + 64 + 64).Take(128).ToArray();
+            Assert.Equal(expectedD, privKey.D);
+            Assert.True(key.Skip(20 + 128 + 64 + 64 + 64 + 64 + 64 + 128).ToArray().Length == 0);
```

Comment:
I think it may help. Lot of magic numbers here. May help to put them into constants so we have an idea what they mean a few years from now.